components/Vote/utils/createVoteHandler.ts

```diff
@@ -1,78 +1,99 @@
 import { UPVOTE, DOWNVOTE } from "~/config/constants";
 import API from "~/config/api";
-import { AuthorProfile, UnifiedDocument } from "~/config/types/root_types";
+import {
+  AuthorProfile,
+  CreatedBy,
+  GrmVoteType,
+  ID,
+  RhDocumentType,
+} from "~/config/types/root_types";
+import { SyntheticEvent } from "react";
+import {
+  emptyFncWithMsg,
+  isNullOrUndefined,
+  nullthrows,
+} from "~/config/utils/nullchecks";
+import { buildGrmVoteApiUri } from "~/config/utils/buildGrmVoteApiUri";
 
 type Args = {
-  voteType: string,
-  unifiedDocument: UnifiedDocument,
-  currentAuthor: AuthorProfile,
-  selectedUserVote?: string,
-  onSuccess: Function,
-  onError: Function,
-}
-
-const getVoteUrl = ({ voteType, unifiedDocument }) => {
-  if (unifiedDocument?.documentType === "post") {
-    if (voteType === UPVOTE) {
-      return API.RH_POST_UPVOTE(unifiedDocument.document.id); 
-    }
-    else if (voteType === DOWNVOTE) {
-      return API.RH_POST_DOWNVOTE(unifiedDocument.document.id); 
-    }
-  }
-  else if (unifiedDocument?.documentType === "paper") {
-    if (voteType === UPVOTE) {
-      return API.UPVOTE(unifiedDocument.documentType, unifiedDocument.document.id); 
-    }
-    else if (voteType === DOWNVOTE) {
-      return API.DOWNVOTE(unifiedDocument.documentType, unifiedDocument.document.id); 
-    }
-  }
-}
+  currentAuthor: AuthorProfile;
+  currentVote: GrmVoteType | null | undefined;
+  commentPayload?: {
+    commentID?: string;
+    commentType?: string;
+    replyID?: string;
+    threadID?: string;
+  };
+  documentCreatedBy: any;
+  documentID: ID;
+  documentType: RhDocumentType;
+  onError: Function;
+  onSuccess: ({
+    increment,
+    voteType,
+  }: {
+    increment: number;
+    voteType: GrmVoteType;
+  }) => void;
+  voteType: GrmVoteType;
+};
 
 export const createVoteHandler = ({
-  voteType,
-  unifiedDocument,
+  commentPayload,
   currentAuthor,
-  selectedUserVote,
+  currentVote,
+  documentCreatedBy,
+  documentID,
+  documentType,
   onError,
   onSuccess,
+  voteType,
 }: Args) => {
-  const voteStrategies = {
-    [UPVOTE]: {
-      increment: 1,
-      url: getVoteUrl({ voteType, unifiedDocument }),
-    },
-    [DOWNVOTE]: {
-      increment: -1,
-      url: getVoteUrl({ voteType, unifiedDocument }),
-    },
-  };
-
-  const { increment, url } = voteStrategies[voteType];
-
-  const handleVote = async () => {
-    const response = await fetch(url, API.POST_CONFIG()).catch(
-      (error) => {
-        console.log("Failed to cast vote", error);
-      }
-    );
-
-    return onSuccess({ voteType, increment });
-  };
-
-  return async (e) => {
-    e.stopPropagation();
+  const formattedDocumentType =
+    documentType === "post" ? "researchhub_posts" : documentType;
+  const resolvedDocumentAuthorID =
+    documentCreatedBy?.authorProfile?.id ||
+    documentCreatedBy?.author_profile?.id;
 
-    if (!currentAuthor?.id) {
-      console.log("Not logged in");
+  return async (event: SyntheticEvent) => {
```

Comment:
The changes here appear to be solid. Maybe we can test on staging first? Thanks for adding these in